src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -241,6 +241,73 @@ export default class Fms {
         return routeSegments.join(DIRECT_ROUTE_SEGMENT_SEPARATOR);
     }
 
+    /**
+     * Return the next waypoint which has an altitude restriction
+     *
+     * @for Fms
+     * @property nextAltitudeRestrictedWaypoint
+     * @type {WaypointModel}
+     */
+    get nextAltitudeRestrictedWaypoint() {
+        const waypoints = this.getAltitudeRestrictedWaypoints();
+
+        return waypoints[0];
+    }
+
+    /**
+     * Return the next waypoint which has an "AT" altitude restriction
+     *
+     * @for Fms
+     * @property nextHardAltitudeRestrictedWaypoint
+     * @type {WaypointModel}
+     */
+    get nextHardAltitudeRestrictedWaypoint() {
+        const waypoints = this.getAltitudeRestrictedWaypoints()
+            .filter((waypoint) => waypoint.altitudeMaximum === waypoint.altitudeMinimum);
+
+        return waypoints[0];
+    }
+
+    /**
+     * Return the next waypoint which has an "AT" speed restriction
+     *
+     * @for Fms
+     * @property nextHardSpeedRestrictedWaypoint
+     * @type {WaypointModel}
+     */
+    get nextHardSpeedRestrictedWaypoint() {
```

Comment:
My thought in general (whether right or wrong) has just been that if it _does something_, it's a method, if it _retrieves or calculates_ something (and sets nothing), it's a getter. Seemed the logical division to me. If we could lay out where a good line between the two is, I've got no issues reorienting myself to align with a different convention.